frontend/review/src/components/GraphBox.js

```diff
@@ -0,0 +1,501 @@
+import React, { useState, useEffect } from "react";
+import styled from "styled-components";
+import { defaults, Scatter } from "react-chartjs-2";
+
+let addDropDate = [];
+
+const LoadingContainer = styled.div`
+  display: flex;
+  width: 100%;
+  height: 100%;
+  justify-content: center;
+  align-items: center;
+  flex-direction: column;
+`;
+
+const ChartTitle = styled.div`
+  font-size: 18px;
+  font-weight: 400;
+  margin-bottom: 5px;
+`;
+
+const ChartDescription = styled.p`
+  font-size: 12px;
+  font-weight: normal;
+  color: #b2b2b2;
+  margin-bottom: 10px;
+`;
+
+const EmptyGraphContainer = styled.div`
+  padding: 10px;
+  text-align: center;
+  color: #8a8a8a;
+`;
+
+const GraphColumn = styled.div`
+  padding: 15px;
+  padding-top: 0px;
+  padding-bottom: 0px;
+  position: relative;
+  width: 100%;
+  flex: 0 0 100%;
+  height: 100%;
+
+  @media (min-width: 768px) {
+    max-width: 50%;
+    flex: 0 0 50%;
+    padding-bottom: 30px;
+  }
+`;
+
+const GraphRow = styled.div`
+  padding: 0px 15px;
+  margin: 0;
+  display: flex;
+  flex-wrap: wrap;
+`;
+
+const GraphContainer = styled.div`
+  padding: 35px;
+  background-color: #ffffff;
+  box-shadow: 0 0 14px 0 rgba(0, 0, 0, 0.07);
+  margin-bottom: 30px;
+  min-height: 500px;
+  flex: 1;
+`;
+
+const genAverageData = (seriesData) => {
+  let averageData = [];
+  let windowSize = 0.05;
+  for (let i = 0; i < seriesData.length; i++) {
+    let xVal = seriesData[i][0];
+    let total = 0;
+    let numInTotal = 0;
+    let j = i;
+    while (j >= 0 && seriesData[j][0] >= xVal - windowSize) {
+      total += seriesData[j][1];
+      numInTotal += 1;
+      j -= 1;
+    }
+    let movingAverageVal = total / numInTotal;
+    averageData.push({
+      x: (xVal * 100).toFixed(2),
+      y: movingAverageVal.toFixed(2),
+    });
+  }
+
+  return averageData;
+};
+
+//PCA Demand Chart Data
+const genDemandChartData = (data, averageData) => {
+  return {
+    datasets: [
+      {
+        type: "line",
+        label: "5% Moving Average",
+        data: averageData,
+        borderColor: "#6378E9",
```

Comment:
someone should make a card to extract colors out into a colors constants file somewhere